lib/ansible/modules/cloud/cloudstack/cs_instance_facts.py

```diff
@@ -265,9 +265,20 @@ def get_result(self, instance):
                     affinity_groups.append(affinitygroup['name'])
                 self.result['affinity_groups'] = affinity_groups
             if 'nic' in instance:
+                network_ip_addresses = {}
                 for nic in instance['nic']:
+                    networkname = nic['networkname']
+                    if networkname not in network_ip_addresses:
+                        network_ip_addresses[networkname] = []
                     if nic['isdefault'] and 'ipaddress' in nic:
                         self.result['default_ip'] = nic['ipaddress']
+                    if 'ipaddress' in nic:
+                        network_ip_addresses[networkname].append(nic['ipaddress'])
+                    for secondaryip in nic['secondaryip']:
```

Comment:
This would look more elegant:
```python
import itertools
...
for nic_section in itertools.chain((nic, ), nic['secondaryip']):
    try:
        ip_address = nic_section['ipaddress']
    except KeyError:
        continue
    network_ip_addresses[networkname].append(ip_address)
```

Reply:
BTW, if `secondaryip` might be missing, replace `nic['secondaryip']` with `nic.get('secondaryip', ())`